roxctl/common/flags/config_file.go

```diff
@@ -0,0 +1,139 @@
+package flags
+
+import (
+	"os"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/pointers"
+)
+
+type Instance struct {
+	ProfileName       string `json:"name"`
+	CaCertificatePath string `json:"caCertPath"`
+	ApiTokenFilePath  string `json:"apiTokenPath"`
```

Comment:
One may need to use different tokens when connecting to the same instance. I suggest adding another level of indirection by extracting credentials (token and password) to a separate structure. Something like:
```go
type Credential struct {
    Name string
    Password string
    ApiToken string
}

type Context struct {
    Name string `json:"name"`
    Context struct{
        InstanceName string `json:"instance"`
        CredentialName string `json:"credential,omitempty"`
    } `json:"context"`
}
```

Please see the kubeconfig structure for inspiration.

Reply:
Gotcha, sounds good. I've been eyeing the `kubeconfig` source code for a bit, and I was debating how closely I should follow it, but all of this makes sense.

I just finished exams, so I'll be working more on this throughout this week!

For now-- should is it essential for me to start adding the other connection parameters for this initial version, as per @parametalol? I was originally planning on working on just getting a baseline merged and extend in future PRs

Reply:
Please stick to the minimal set of options. We'd like to do some refactoring to simplify the priority layering of the configuration arguments. So that there is a single source of default values (preferably as an instance of the Instance object), and then env vars and CLI flags to reference them.

Reply:
Sounds good, does the team agree that the precedence order should be flag > env > config > default?

Reply:
Yes.

From my quick test, the YAML library allows the object, passed to the `Unmarshal()` function, to have predefined values, which won't be overridden by empty input (assuming `omitempty`).
Please consider testing and using this functionality for defining the single source of default configuration values.